module/VuFind/src/VuFind/Search/Factory/LibGuidesAZBackendFactory.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Factory for LibGuides A-Z Databases backends.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2013.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Search\Factory;
+
+use VuFindSearch\Backend\LibGuidesAZ\Backend;
```

Comment:
If you want to reduce the number of spare classes you're creating, I suspect that you can eliminate all of the LibGuidesAZ classes, and just reuse equivalent LibGuides classes. Having separate classes is important inside the VuFind module since it relates to routing and parameter processing, but once you get to the search backend code, there's no consequence to reusing the same backend code multiple times (we already do this for Solr, etc.).

It looks to me like the only code that's unique to the Backend\LibGuidesAZ classes is the widget_type used in the query builder... and you could address that by adding a `defaultWidgetType` property to the class, and a `setDefaultWidgetType` method, and then have the factory set up the query builder appropriately.

There's nothing wrong with your current approach if you think there might be a reason to more explicitly differentiate functionality in the future, but if you want to reduce the number of files in the PR, I think this could be an easy win.

Reply:
I took this advice.  I'm not sure about whether more differentiation will be necessary in the future, but I think better to cross that bridge later than add five trivial files at this point.